docs/se/guides/migration_4x.adoc

```diff
@@ -0,0 +1,216 @@
+
+There is no more backward compatibility with Helidon SE version 3 and earlier! All reactive code written using Reactive paradigm should be redesigned and rewritten in blocking paradigm.
+
+== Server initialization and start up
+
+In Helidon 1.x-3.x to star a server the following actions should be done:
+
+[source, java]
+.Start Helidon SE 3.x Server
+----
+static Single<WebServer> startServer() {
+
+    Config config = Config.create();
+
+    WebServer server = WebServer.builder(createRouting(config))
+            .config(config.get("server"))
+            .addMediaSupport(JsonpSupport.create())
+            .build();
+
+    Single<WebServer> webserver = server.start();   <1>
+
+    webserver.thenAccept(ws -> {    <2>
+                System.out.println("WEB server is up! http://localhost:" + ws.port() + "/greet");
+                ws.whenShutdown().thenRun(() -> System.out.println("WEB server is DOWN. Good bye!"));
+            })
+            .exceptionallyAccept(t -> { <3>
+                System.err.println("Startup failed: " + t.getMessage());
+                t.printStackTrace(System.err);
+            });
+
+    return webserver;
+}
+----
+<1> Server is started in asynchronous way. A `Single` object is returned.
+<2> Wait for the Server to start and print the message in asynchronous way.
+<3> Gracefully handle exceptions if they occur during the initialization process.
+
+Since Helidon SE is reactive, during the start a `Single` object is returned, the server has been started in asynchronous way. We have to use reactive methods like `thenAccept` to wait for the server to start and the to perform the desired action. The exception handling should also be done in reactive way using the corresponding method.
+
+In Helidon there are no more Reactive APIs, so the Server is started much simpler:
+
+[source, java]
+.Start Helidon SE 4.x Server
+----
+public static void main(String[] args) {
+
+    Config config = Config.create();
+    Config.global(config);
+
+    WebServer server = WebServer.builder()  <1>
+            .config(config.get("server"))
+            .routing(Main::routing)
+            .build()
+            .start();   <2>
+
+    System.out.println("WEB server is up! http://localhost:" + server.port() + "/greet");   <3>
+}
+----
+<1> Configure the Server.
+<2> Start the Server. No reactive objects returned.
+<3> Print a message when the Server is started.
+
+Just create it, configure it, and wait for it to start. The "waiting" time is now managed by the Java Virtual Machine. If waiting time is too long since the app is waiting for some resource, another task may be assigned to the current thread, to do something useful. Then when server is ready, it will notify the scheduler, whit will mount it on a free carrier thread.
```

Comment:
I think you can remove "The waiting time...free carrier thread."  I don't think we need to explain how virtual thread scheduling works in a migration guide.

Reply:
oki)